src/libYARP_math/src/RandScalar.cpp

```diff
@@ -45,13 +44,17 @@ RandScalar::~RandScalar()
 
 double RandScalar::get()
 {
-    return implementation(impl)->random();
+    std::mt19937 gen =  *(implementation(impl));
+    double ret = gen()/gen.max();
```

Comment:
Are you sure this is working? `gen()` and `gen.max()` return integers, so I am afraid this division will always return the integer 0, that is cast to 0.0. The use of static_cast I was referring in the other comment would be: 
~~~
double ret = static_cast<double>(gen())/static_cast<double>(gen.max());
~~~

Reply:
yep you're right, don't know how it could have been passing tests, for sure I did it the wrong way.
Also explicitly recreating an mt19937 was a not-to-do thing. I fixed it in a readable way here https://github.com/robotology/yarp/pull/1444/commits/4a3e942e7d898bc243b0c05cfb987aa6502cd709

Reply:
I suspect that a test should be added to verify that the get is not always returning the same number, otherwise we are prone to this kind of implementations: https://xkcd.com/221/ . :D

Reply:
That was exactly my previous implementation :laughing: the thing is there already are tests for the random generation, I was just not able to run them right.

Reply:
Mhh, also Travis was green regarding the tests. Are we sure that the current tests are covering this case on this class?

Reply:
https://github.com/robotology/yarp/blob/master/tests/libYARP_math/randTest.cpp should cover this, when i run it via `harness` some of the tests were failing because they were (indirectly) calling the `RandScalar::get()`. But when I run it via `make test` all math tests gave 'passed' as result.

Reply:
Mhh, the problem is that also Travis is running the tests via `make test` (or the `ctest` command https://github.com/robotology/yarp/blob/master/.travis.yml#L389, that is equivalent). This means that we are vulnerable to anything that breaks these tests, because Travis will not warn us.
You can check the actual command run by ctest by executing:
~~~
ctest -VV -R NameOfTheTestYouWantToRun
~~~
and see how this differs from the command that you run via `harness`.

Reply:
Ok, sorry for the delay, I was not at work these days :sweat_smile:
Found out that there was a problem with the test names and two math tests were not run at all. Fixed in https://github.com/robotology/yarp/pull/1444/commits/82ad60eade5a0299f6ca9705d30d0182300f807e